contracts/JBController.sol

```diff
@@ -821,6 +882,8 @@ contract JBController is IJBController, JBTerminalUtility, JBOperatable, Reentra
         );
 
     // Set overflow allowances if there are any.
+    // The routine makes an assumption that `distributionLimitOf` changes with each reconfiguration
+    // much more frequently than `overflowAllowanceOf` or `currencyOf`. The data structures and storage patterns are thus different.
```

Comment:
Perhaps some analysis or polls to JB project owners can confirm these intuitions. That being said, I do believe that the distributionLimits are definitely prone to change from cycle to cycle.